src/main/java/com/google/ehub/servlets/UserDataManager.java

```diff
@@ -0,0 +1,25 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+
+public class UserDataManager {
+  public static UserProfile getUserProfile(String email) {
+    Filter propertyFilter = new FilterPredicate(EMAIL_PROPERTY_KEY, FilterOperator.EQUAL, email);
```

Comment:
I don't know if Users API returns emails in lowercase but I would try to make sure that all the emails in Datastore are in lowercase so the equality comparison doesn't cause some sort of bug.

Reply:
I will test to see if it does